pkg/syncer/syncertest/fake/dynamic_client.go

```diff
@@ -219,6 +225,13 @@ func (dc *DynamicClient) patch(action clienttesting.Action) (bool, runtime.Objec
 	if err := incrementResourceVersion(patchedObj); err != nil {
 		return true, nil, fmt.Errorf("failed to increment resourceVersion: %w", err)
 	}
+	if found {
+		if err = updateGeneration(cachedObj, patchedObj, dc.scheme); err != nil {
+			return true, nil, fmt.Errorf("failed to update generation: %w", err)
+		}
+	} else {
+		patchedObj.SetGeneration(1)
```

Comment:
`initGeneration`?

Reply:
initGeneration allows the user to specify a generation, while just setting it overrides any value specified by the caller.

I deliberately only used initGeneration in the Create call (which the dynamic client doesn't have implemented ATM).

I'm not really sure if Update should allow creation or not, but I assume it should at least for SSA.

Normally, I think the apiserver would always set Generation, and ignore what the user specified on Create, but I wanted to allow users of the fake client to be able to set their own Generation for testing purposes, but only on Create.

We should probably add Create for the dynamic client, with similar behavior, but we haven't needed it for tests yet.

If you think Updates that create a new object should allow the user to set the Generation tho, let me know. It kinda depends on whether we want to follow the apiserver behavior or allow different behavior because we need it for testing.